codeship-services.yml

```
@@ -26,6 +28,17 @@ testing:
   encrypted_dockercfg_path: dockercfg.encrypted
   default_cache_branch: 'dev'
 
+e2e:
+  build:
+    dockerfile: e2e.Dockerfile
+  encrypted_env_file: ./ci-env-vars.encrypted
+  cached: true
+  encrypted_dockercfg_path: dockercfg.encrypted
+  default_cache_branch: 'dev'
```

```suggestion
  default_cache_branch: 'e2e'
```
The cache probably won't work from `dev` as dev won't run e2e and build from that Dockerfile at all, so instead we could keep an `e2e` branch semi-regularly updated from dev? As a performance optimisation only, so wouldn't be too important if we forgot to keep it updated. Either that or we could just make `dev` build the e2e docker container, even if it doesn't use it.

---

I see, that makes a lot of sense!

By the way , currently the https://e2e.tupaia.org ("default" baseline branch) is a production clone. I know that testers usually use `dev` to test stuff midweek, but before a release we use production as a baseline.

What do you think? In that case, we can regularly update `e2e` from `master` (instead of `dev`). We can see how this goes and maintain two permanent baseline branches, if need be (https://e2e.tupaia.org, https://dev-e2e.tupaia.org).

---

Happy for `e2e` to be regularly updated from `master` :-)